include/alpaka/acc/Tag.hpp

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2022 Simeon Ehrig
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/BoostPredef.hpp>
+
+#include <iostream>
+#include <type_traits>
+
+#define CREATE_ACC_TAG(tag_name)                                                                                      \
+    struct tag_name                                                                                                   \
+    {                                                                                                                 \
+        static std::string get_name()                                                                                 \
+        {                                                                                                             \
+            return #tag_name;                                                                                         \
+        }                                                                                                             \
+    }
+
+namespace alpaka
+{
+    CREATE_ACC_TAG(AccCpuFibersTag);
```

Comment:
IMO this should be in the namespace `tag` then there is no need to add `Tag` to the name.
IMO namespaces are exactly for such hierarchy collections.

The usage will not be much longer `alpaka::AccCpuFibersTag` vs `alpaka::tag::AccCpuFibers`.
Or with an inline namespace `alpaka::AccCpuFibers` but this will collide with the implementation of the accelerator :-(

I am not sure if there is a need to name the tag `Acc...`, `alpaka::tag::CpuFibers` is IMO fine.

Reply:
I would suggest two solutions:

1. I would cover @psychocoderHPC  suggestion and provide only the tags inside the namespace `tag` without the `Acc` prefix (e.g. `alpaka::tag::CpuFibers`). A flat namespace would be not useful (e.g. `alpaka::CpuFibers`), because the name is to meaningless.
2. Prefix the names with `Tag` instead `Acc`, e.g. (`alpaka::TagCpuFibers`). We can also put the tags in the namespace `tag` for a better organization. The full API is only a little bit verbose/redundant: `alpaka::tag::TagCpuFibers`

Reply:
I vote for option 2 without the extra namespace, so `alpaka::TagCpuFibers`. @bernhardmgruber's mental health is important to me and he put so much work into getting rid of additional namespaces in alpaka... but on a more serious note I think it matches our current `AccXyz` naming scheme nicely.